server/webapp/WEB-INF/rails.new/spec/lib/message_verifier_spec.rb

```diff
@@ -0,0 +1,39 @@
+##########################################################################
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+require 'spec_helper'
+
+describe MessageVerifier do
+  describe :verify do
+    it "should check if the message is tampered" do
+      message = 'message'
+      signature = 'message_signature'
+      public_key_path = '/config/update_server_key.pub'
+      public_key = 'update_server_public_key'
+      decoded_public_key = 'decoded_public_key'
+      digest = double('OpenSSL::Digest::SHA512')
+      pkey = double('OpenSSL::PKey::RSA')
+
+      File.should_receive(:read).with(public_key_path).and_return(public_key)
+      OpenSSL::Digest::SHA512.should_receive(:new).and_return(digest)
+      Base64.should_receive(:decode64).with(public_key).and_return(decoded_public_key)
+      OpenSSL::PKey::RSA.should_receive(:new).with(decoded_public_key).and_return(pkey)
+      pkey.should_receive(:verify).with(digest, signature, message)
```

Comment:
I don't see a reason to actually test what's being invoked, we already have a public key, we can test actual verification with a good and bad message/digest.

I feel that's a much "safer" test in case things break.